server/store/sqlstore/store.go

```diff
@@ -710,15 +698,24 @@ func (s *SQLStore) SaveChatSubscription(subscription storemodels.ChatSubscriptio
 	return nil
 }
 
-func (s *SQLStore) SaveChannelSubscription(tx *sql.Tx, subscription storemodels.ChannelSubscription) error {
+func (s *SQLStore) SaveChannelSubscription(subscription storemodels.ChannelSubscription) error {
+	tx, err := s.db.Begin()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		_ = tx.Rollback()
```

Comment:
I /think/ this is fine, but wondering if we should be explicit and copy the pattern in the server to not rollback unless necessary?

```go
// finalizeTransactionX ensures a transaction is closed after use, rolling back if not already committed.
func finalizeTransactionX(transaction *sqlxTxWrapper, perr *error) {
	// Rollback returns sql.ErrTxDone if the transaction was already closed.
	if err := transaction.Rollback(); err != nil && err != sql.ErrTxDone {
		*perr = merror.Append(*perr, err)
	}
}
```

(Haven't looked at the `merror` stuff, 0/5 on that.)

Reply:
That pattern is not "rollback unless is necessary", that patten ensure that if there is a problem with the rollback, it is added to a list of errors, to be handled outside of the function. In this case, I think we can't do much with the error there, we can maybe log it, but I don't think we have much to do. If commit went wrong, and rollback went wrong... what it means? If commit went ok, and rollback went wrong, what it means? I think just swallowing that error, or logging it is enough in this case.

Reply:
Ah, you're right: I forgot that the `transaction.Rollback()` is called unconditionally in both cases. Never mind :D